### ISSUE TITLE
Rule explicit_self does not report violations

### ISSUE THREAD

**Issue by gereons** [realm/SwiftLint#2974]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

I'm running `swiftlint analyze` with rule `explicit_self` enabled in my project:

````
xcodebuild clean -workspace XYZ.xcworkspace -scheme XYZ
xcodebuild -workspace XYZ.xcworkspace -scheme XYZ >xcodebuild.log
swiftlint analyze --compiler-log-path xcodebuild.log
````

Snippet from `.swiftlint.yml`:

````
analyzer_rules:
 - explicit_self
````

and get "0 violations found", although there are a few.

The same `.swiftint.yml` file works in a newly created dummy project where I've added an intentional violation. I am unable to share the project code, unfortunately, is there anything I can do help narrow this down?

Swiftlint version: 0.37.0, installed via homebrew
Xcode version: 11.2.1

**Comment by jpsim** [comment]
> I am unable to share the project code, unfortunately, is there anything I can do help narrow this down?

If you can create a separate project that reproduces the issue that you _can_ share, that will go a long way towards helping identify why this isn't working for you.

**Comment by gereons** [comment]
I tried that initially but was unable to reproduce the issue at the time.

I did however manage to spend some more time on this today and was able to find two unrelated issues surrounding the analyzer and `explicit_self` which I'll post separately.

**Comment by jpsim** [comment]
Thanks for investigating. Looking forward to seeing what you found.

**Comment by gereons** [comment]
See #3020 and #3021.
